Workspace: cargo clippy --all-features fails on integration_routing_rendering placeholder feature in zfb-render

## Found during

Workflow `/x-wt-teams` for epic #27 (MDX React Components) — confirmed pre-existing on `main` both before and after merge of PR #36.

## Description

`cargo clippy --workspace --all-features --all-targets` fails to compile the gated test `crates/zfb-render/tests/integration_routing_rendering.rs`:

```
error[E0432]: unresolved imports `zfb_render::render_route`, `zfb_render::Framework`, `zfb_render::RenderInput`, `zfb_render::RenderOutput`
error[E0432]: unresolved import `zfb_router`
error: could not compile `zfb-render` (test "integration_routing_rendering") due to 2 previous errors
```

The test is intentionally feature-gated (`#![cfg(feature = "integration_routing_rendering")]`) because the orchestrator API surface (`render_route`, `RenderInput`, `RenderOutput`) and the `zfb-router` workspace dep have not yet landed. So the test is correctly disabled by default — but enabling `--all-features` (which CI tooling, IDE checks, and developer-side `cargo check --all-features` all do) breaks the build.

## Repro

```bash
cargo clippy --workspace --all-features --all-targets
```

## Suggested fix

Either:
1. **Remove the feature stub** — drop `integration_routing_rendering = []` from `crates/zfb-render/Cargo.toml`, leave the test source on disk with a TODO comment explaining it cannot be compiled until the orchestrator API lands. (Most direct.)
2. **Make the deps optional + feature-gated** — add `zfb-router = { path = "../zfb-router", optional = true }` and gate the unresolved imports behind the same feature in the source crate. (Bigger surface; only worth it if the API is close to landing.)
3. **Move the test out of the workspace until ready** — relocate to a placeholder doc file, restoring it once the orchestrator API ships.

Option 1 is cleanest unless the orchestrator work is imminent.

---

*Discovered during `/x-wt-teams` workflow on epic #27 — not related to that epic's changes.*

--- a/crates/zfb-render/Cargo.toml
+++ b/crates/zfb-render/Cargo.toml
@@ -13,11 +13,6 @@ description = "zfb render: SWC TSX→JS pipeline + JS runtime + page render orch
 # (trait + SWC pipeline + orchestrator) compiles and tests without it.
 default = []
 deno_core_host = ["dep:deno_core", "dep:tokio"]
-# Gates the end-to-end routing+rendering integration test. Off until the
-# orchestrator exposes render_route + RenderInput/Output and zfb-render
-# depends on zfb-router. See tests/integration_routing_rendering.rs for
-# the fixtures and harness already in place.
-integration_routing_rendering = []
 
 [dependencies]
 # Sibling crate: MDX → JSX-source emitter. The loader funnels `.mdx` and

--- a/crates/zfb-render/tests/integration_routing_rendering.rs
+++ b/crates/zfb-render/tests/integration_routing_rendering.rs
@@ -1,3 +1,23 @@
+// TODO(parked): This integration test is parked.
+//
+// It depends on the orchestrator API surface (`render_route`,
+// `RenderInput`, `RenderOutput`) and the `zfb-router` workspace crate,
+// none of which exist yet. Until those land, the test cannot compile,
+// and a `cargo clippy --workspace --all-features --all-targets` (or
+// `cargo check --all-features`) run otherwise breaks the workspace
+// build.
+//
+// To work around that, this file is unconditionally excluded from
+// compilation via `#![cfg(any())]` (the canonical "always-false" cfg)
+// and the matching `integration_routing_rendering = []` feature has
+// been removed from `crates/zfb-render/Cargo.toml`. The test source,
+// fixtures, layouts, components, and harness are intentionally kept on
+// disk so they remain version-controlled.
+//
+// When the orchestrator API and `zfb-router` crate land, restore the
+// original `#![cfg(feature = "integration_routing_rendering")]`
+// attribute below and re-add the feature line in `Cargo.toml`.
+
 //! End-to-end integration tests for the Epic 3 routing + rendering
 //! pipeline.
 //!
@@ -44,7 +64,7 @@
 // can be turned on with a single Cargo.toml change once the orchestrator
 // exposes the expected entry points and zfb-render adds zfb-router as
 // a dep. Tracked as the natural follow-up to Epic 3 (Subs 1–7).
-#![cfg(feature = "integration_routing_rendering")]
+#![cfg(any())]
 #![allow(dead_code)]
 
 use std::path::{Path, PathBuf};

--- a/docs/src/components/language-switcher.astro
+++ b/docs/src/components/language-switcher.astro
@@ -4,6 +4,13 @@ import { defaultLocale, type Locale } from "@/config/i18n";
 
 interface Props {
   lang?: Locale;
+  /**
+   * Optional list of supported locale codes. The component derives its links
+   * from `Astro.url.pathname` and the locale config in settings, so this prop
+   * is currently unused but accepted to keep the call site in `header.astro`
+   * type-safe (it forwards the project-wide `locales` array from `i18n.ts`).
+   */
+  locales?: readonly Locale[];
 }
 
 const { lang = defaultLocale } = Astro.props;

--- a/docs/src/config/settings.ts
+++ b/docs/src/config/settings.ts
@@ -6,6 +6,8 @@ export type {
   LocaleConfig,
   VersionConfig,
   FooterConfig,
+  FrontmatterPreviewConfig,
+  TagPlacement,
 } from "./settings-types";
 import type {
   HeaderNavItem,
@@ -14,6 +16,9 @@ import type {
   LocaleConfig,
   VersionConfig,
   FooterConfig,
+  FrontmatterPreviewConfig,
+  TagPlacement,
+  TagGovernanceMode,
 } from "./settings-types";
 
 export const settings = {
@@ -50,6 +55,27 @@ export const settings = {
   sidebarResizer: true as boolean,
   sidebarToggle: true as boolean,
   htmlPreview: undefined as HtmlPreviewConfig | undefined,
+  /**
+   * Optional URL of the project's GitHub repository. When set, the header
+   * renders a GitHub link icon. Leave undefined to omit the link.
+   */
+  githubUrl: undefined as string | undefined,
+  /**
+   * Frontmatter preview block, rendered above the doc body.
+   * `false` disables the feature entirely. An object enables it and may
+   * customise which keys are hidden via `ignoreKeys` / `extraIgnoreKeys`.
+   */
+  frontmatterPreview: false as false | FrontmatterPreviewConfig,
+  /**
+   * Where the per-page tag list is rendered. Undefined hides it.
+   * - `"after-title"` — directly below the page title.
+   * - `"before-pager"` — directly above the prev/next pager at the foot.
+   */
+  tagPlacement: undefined as TagPlacement | undefined,
+  /** Master on/off switch for consulting `tag-vocabulary.ts` at runtime. */
+  tagVocabulary: false as boolean,
+  /** Tag governance enforcement level. See `TagGovernanceMode` for values. */
+  tagGovernance: "off" as TagGovernanceMode,
   versions: [] as VersionConfig[],
   claudeResources: {
     claudeDir: ".claude",

